{{ cookiecutter.package_name }}/.github/workflows/tox-tests.yml

```diff
@@ -0,0 +1,104 @@
+# GitHub Actions workflow for testing and continuous integration.
+#
+# This file performs testing using tox and tox.ini to define and configure the test environments.
+
+name: Python Tests
+
+on: [push, pull_request]
+
+jobs:
+
+  # Set up matrix to run tox tests across lists of os, python version, and tox environment
+  matrix_tests:
+    runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"  # github actions doesn't yet support this natively so add here
+    strategy:
+      matrix:
+        # Github actions supports ubuntu, windows, and macos virtual environments:
+        # https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+        #
+        # Only run on ubuntu by default, but can add other os's to the test matrix here.
+        # For example -- os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest]
+
+        # Test python 3.7 and 3.8 by default.
+        python-ver: [7, 8]
+
+        # Specify which tox environments to test in this list.
+        tox-env: [cov, alldeps, devdeps, astropylts]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up python 3.${{ matrix.python-ver }} with tox environment py3${{ matrix.python-ver }}-${{ matrix.tox-env }} on ${{ matrix.os }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.${{ matrix.python-ver }}
+    - name: Install base dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Test with tox
+      run: |
+        tox -e py3${{ matrix.python-ver }}-${{ matrix.tox-env }}
+    #
+    # # This is an example of how to upload coverage to codecov
+    # - name: Upload coverage to codecov
+    #   if: matrix.tox-env == 'cov' && matrix.python-ver == '8'
```

Comment:
@nstarman has an alternative approach at astropy/astropy#11040

Also, I don't think you need a token anymore for Actions.

Reply:
coverage handling also needs fixing up to merge coverage output from multiple tox environments. this hack specifically uploads from one specific environment while the one in astropy uploads from each environment containing `-cov` so, afaict, whatever ends up on codecov at the end is whatever the last one uploaded happened to be. i have bookmarked a recipe for how to combine coverage from multiple tox envs so i'll dig that out and see if i can apply it.

and, indeed, no codecov token is required any longer for public repos.

Reply:
i ended up punting on the complicated path in favor of what astropy and photutils do: just have one test run with coverage enabled and upload those results.